File: wire-android-automation/common/src/main/java/com/wearezeta/auto/common/email/inbucket/InbucketService.java
/*
Copyright 2016 StepStone Services

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
 */

package com.wearezeta.auto.common.email.inbucket;

import com.wearezeta.auto.common.email.inbucket.models.Message;
import com.wearezeta.auto.common.email.inbucket.models.MessageInfo;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Path;

import java.util.List;

interface InbucketService {

    @GET("api/v1/mailbox/{name}")
    Call<List<MessageInfo>> getMailbox(@Path("name") String mailboxName);

    @GET("api/v1/mailbox/{name}/{id}")
    Call<Message> getMessage(@Path("name") String mailboxName, @Path("id") String messageId);

    @GET("api/v1/mailbox/{name}/{id}/source")
    Call<ResponseBody> getMessageSource(@Path("name") String mailboxName, @Path("id") String messageId);

    @DELETE("api/v1/mailbox/{name}")
    Call<String> deleteMailbox(@Path("name") String mailboxName);

    @DELETE("api/v1/mailbox/{name}/{id}/source")
    Call<String> deleteMessage(@Path("name") String mailboxName, @Path("id") String messageId);
}

File: tests/testsCore/src/main/kotlin/com/wire/android/tests/core/models/InbucketService.kt
/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 */
package com.wire.android.tests.core.models

import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Path

interface InbucketService {

    @GET("api/v1/mailbox/{name}")
    fun getMailbox(@Path("name") mailboxName: String): Call<List<MessageInfo>>

    @GET("api/v1/mailbox/{name}/{id}")
    fun getMessage(
        @Path("name") mailboxName: String,
        @Path("id") messageId: String
    ): Call<Message>

    @GET("api/v1/mailbox/{name}/{id}/source")
    fun getMessageSource(
        @Path("name") mailboxName: String,
        @Path("id") messageId: String
    ): Call<ResponseBody>

    @DELETE("api/v1/mailbox/{name}")
    fun deleteMailbox(@Path("name") mailboxName: String): Call<String>

    @DELETE("api/v1/mailbox/{name}/{id}/source")
    fun deleteMessage(
        @Path("name") mailboxName: String,
        @Path("id") messageId: String
    ): Call<String>
}
